src/praisonai-code/praisonai_code/cli/features/interactive_tools.py

```
@@ -193,6 +204,100 @@ def _load_basic_tools() -> Dict[str, Callable]:
+
+    # The core edit engine marks edit_file/apply_patch as high-risk and routes
+    # them through the approval registry. In the interactive "auto" approval
+    # mode the agent runs with full privileges (matching the ACP tools), so
+    # register these as context-approved to avoid a blocking console prompt.
+    # Manual/scoped modes leave the normal approval flow (with diff preview)
+    # in place.
+    if config.approval_mode == "auto":
+        try:
+            from praisonaiagents.approval import set_yaml_approved_tools
+            set_yaml_approved_tools(["edit_file", "apply_patch"])
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)